### PR TITLE
translate mode: support x11 in client

### PR DESCRIPTION
**Pull request by chiehw**
https://github.com/rustdesk/rustdesk/issues/2549

### CODE REVIEW

File: src/keyboard.rs [hunk 1]
```diff
@@ -833,6 +833,39 @@ pub fn translate_key_code(peer: &str, event: &Event, key_event: KeyEvent) -> Opt
...
+    #[cfg(target_os = "macos")]
+    // ignore right option key
+    if event.platform_code as u32 == rdev::kVK_RightOption {
+        true
+    } else {
+        false
+    }
+
+    #[cfg(target_os = "windows")]
+    if unsafe { IS_0X021D_DOWN } {
```
Comment (fufesou) [inline src/keyboard.rs]: `if unsafe { IS_0X021D_DOWN } && event.position_code == 0xE038` may be more clear.

File: src/keyboard.rs [hunk 2]
```diff
@@ -849,27 +882,17 @@ pub fn translate_keyboard_mode(peer: &str, event: &Event, key_event: KeyEvent) -
         }
     }
 
-    #[cfg(target_os = "macos")]
-    // ignore right option key
-    if event.code as u32 == rdev::kVK_RightOption {
+    if is_altgr(event) {
```
Comment (fufesou) [inline src/keyboard.rs:885]: `is_altgr` may not fit macOS?

Comment (fufesou) [inline src/keyboard.rs:881]: `is_altgr` may not fit macOS. The following code seems more clear.
```rust
    #[cfg(target_os = "macos")]
    // ignore right option key
    if event.code as u32 == rdev::kVK_RightOption {
        return events;
    }
    #[cfg(not(target_os = "macos"))]
    if is_altgr(event) {
        return events;
    }
```

**Comment by rustdesk** [comment]
above conservation resolved?

**Comment by chiehw** [comment]
> above conservations resolved?

Yes

**Pull request merged**